CMakeLists.txt

```diff
@@ -167,3 +167,24 @@ function(mbed_generate_bin_hex target)
         VERBATIM
     )
 endfunction()
+
+#
+# Configure Mbed OS CMake application (default).
+#
+function(mbed_configure_application target)
+    mbed_set_language_standard(${target})
+    mbed_set_mbed_target_linker_script(${target})
+    mbed_generate_bin_hex(${target})
+endfunction()
+
+#
+# Configure Mbed OS CMake component (default).
+#
+function(mbed_configure_component component)
```

Comment:
You are really passing a target, so I suggest to rename the argument to reflect that.
```suggestion
function(mbed_configure_component target)
```

Reply:
wouldn't it be a component. if I created nfc component, I would pass nfc here. It is target per se but this is usable only for Mbed OS components.

Reply:
what about `target_component` or `component_target`? ;)

Reply:
fixed,  renamed the arguments

Reply:
It is still just a target that is passed right? So let's just call it that, `target`. The function name already explains what you do with the target.